### ISSUE TITLE
`Implementation-Version: vunspecified` returned under edge conditions

### ISSUE THREAD

**Issue by Dispader** [Dispader/manifesto#26]
When

* no `.git` repository exists for the project and
* no `projectRoot.version` is specified and
* the `com.jfrog.bintray` plugin is included for a client build

`Implementation-Version: vunspecified` is set in the manifest.

**Comment by Dispader** [comment]
Hey, @cdinger — any opinion on what the default behavior should be?

Steal the version if it exists and validate that it looks like a version number?

**Comment by Dispader** [comment]
Here's a test `build.gradle`:

```Groovy
plugins {
    id 'java'
    id 'com.github.dispader.manifesto' version '0.4.0'
    id 'com.jfrog.bintray' version '1.7'
}

group = 'com.github.dispader'

artifacts {
    jar
}

bintray {
    user = 'dispader'
    key = 'REDACTED'
    filesSpec {
        from 'build/libs'
        into '.'
    }
    pkg {
        repo = 'maven'
        name = 'manifesto-test'
        userOrg = 'dispader'
        licenses = ['Apache-2.0']
        vcsUrl = 'https://github.com/Dispader/manifesto'
    }
}

**Comment by Dispader** [comment]
Proposed solution so far— only re-write to `v` form if the version is SemVer-parsable.

**Comment by Dispader** [comment]
Well, here's a bit of a game-changer: it [looks like manifest specification and implementation versions shouldn't be prepended with `v`](https://docs.oracle.com/javase/tutorial/deployment/jar/packageman.html).

I think I'm going to have to change the tests and rethink this.

**Comment by cdinger** [comment]
I think it's OK to break in a friendly way if there's no information available from which it can infer a version. I think breaking is better than inventing something in this case—it shows the user that this is meant to be a tool that leans on git. If you're not using git, this tool makes no sense.

I also noticed the `v` issue way back when... should be simple to code around.

**Comment by cdinger** [comment]
I even wonder about handling a `projectRoot.version` if it exists. What does that mean to use Manifesto, but then manually specify a version? Seems like that should kaboom too, no?

**Comment by Dispader** [comment]
@cdinger I could totally get behind us blowing up if `projectRoot.version` is already set— I was going for setting the `projectRoot.version` to match the one determined by Git.

But, I'm waffling a bit: that was mainly to facilitate releasing using things like the Bintray or Artifactory plugin (or publishing Gems).  Hrrrm.

**Comment by Dispader** [comment]
Interesting— yeah, how would we even square the circle, there?  We should probably blow up if:

1. the project isn't Git controlled __or__
1. project.version is set

**Comment by Dispader** [comment]
We seem to finally have some reasonable warning working with `v0.13.1` for projects that aren't Git-controlled.  (Throwing a kill to the build proved ugly— the user couldn't issue `gradle tasks` without a catastrophic-looking stack dump, which seemed unfriendly.)

Now I'm still pondering whether this is closed: I think that the addenda to case '2.' above is still needed: if there's a `projectRoot` which isn't a valid project, we might consider not blowing up.

That said, we should warn for projects with `version` pre-set, and that's not in there yet.

**Comment by Dispader** [comment]
https://github.com/Dispader/manifesto/projects/1#card-254882

**Comment by Dispader** [comment]
Mostly addressed in fe903ea94a834528facb8b90d217de041283982a, but I've noticed something that I don't like: if a project is controlled by Git, but has no tags, the version detector doesn't default a version.  I'm wagering that no human would consider tagging a SemVer alpha project (i.e.: something accurately described as `v0.0.0`), so I'd like that default to make more sense.

Right now, for Git-controlled projects with no default tags, Manifesto warns at load with the message:

>`warning: The Manifesto plugin is only useful for projects using git source control.`

Staying on this particular bug/issue: the message should probably be a different one if there is no base tag.  However: if we've got no data **about** the base tag, that's not particularly... helpful?

I want to Close this resolved and add an issue for retaining the message, but offering an opinionated `0.0.0` as a reported version for untagged projects.
